docs/decorator.md

```diff
@@ -1,46 +1,30 @@
 ## Decorator
 
-The following scenarios are the perfect use cases for choosing a Wallaby decorator:
+The purposes of using a decorator are:
+- [Basic Customization](#basic-customization): to customize how resource should be displayed on index/show/form page via managing metadata.
+- [Advanced Customization](#advanced-customization): to encapsulate view logics for a model. for example, to define a `full_name` method to concatenate `first_name` and `last_name` together and get it to be displayed.
+- [More Metadata Options](#more-metadata-options)
+- [Filters](#filters): predefined filters
+  - [Use Existing Named Scope](#use-existing-named-scope)
+  - [Define a New Scope](#define-a-new-scope)
+  - [More Filter Options](#more-filter-options)
+- [Misc Customization](misc-customization)
+  - [`to_label`](#to_label)
+  - [`primary_key`](#primary_key)
 
-- change what fields to show/list
-- update the metadata information of a field, e.g. title, type
-- add custom field
+First of all, let's see how to create a decorator so that Wallaby can pick it up:
+
+### Declaration
 
 ```ruby
 #!app/decorators/product_decorator.rb
 
 # NOTE: Product in ProductDecorator must be singular
 class ProductDecorator < Wallaby::ResourceDecorator
-  # example of metadata:
-  # {
-  #   column_name: { # minimum data requires type and label
-  #     type: string,
-  #     label: 'Column Name',
-  #     name: 'column_name'
-  #   }
-  # }
-  self.index_fields # metadata for index page
-  self.show_fields  # metadata for show page
-  self.form_fields  # metadata for new/create/edit/update page
-
-  # example of field_names:
-  # [ 'id', 'column1', ..., 'created_at', 'updated_at' ]
-  self.index_field_names  # array of column names to display on index page
-  self.show_field_names   # array of column names to display on show page
-  self.form_field_names   # array of column names to display on form page
-
-  # examples
-  self.index_fields[:email][:type] = 'email'
-  self.index_fields[:full_product_name] = { type: 'string', label: 'Product Name' }
-  self.index_field_names << 'full_product_name'
-
-  def full_product_name
-    [ product_name, sku ].compact.join ' - '
-  end
 end
 ```
 
-Similarly, if the name ProductDecorator is taken, we could use another name and add a method to specify the model class.
+If the name `ProductDecorator` is taken, it is possible to use other name but needs to create a method `self.model_class` to specify the model as below:
```

Comment:
I think the wording below would work better, showing that the method MUST be defined shows how important defining this method is.

> If the name `ProductDecorator` is taken, it is possible to use another name however the `self.method_class` must be defined to specify the model as the example below: